src/compiler/checker.ts

```diff
@@ -16122,7 +16129,7 @@ namespace ts {
 
             if (produceDiagnostics) {
                 checkTypeForDuplicateIndexSignatures(node);
-                checkUnusedTypeParameters(node);

```

Comment:
Should anything use checkUnusedTypeParameters except in a deferred context? It looks like it's still called from a bunch of places.